environs/imagedownloads/simplstreams.go

```diff
@@ -0,0 +1,214 @@
+// Package imagedownloads implements image-downloads metadata from
+// simplestreams.
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package imagedownloads
+
+import (
+	"net/url"
+	"sort"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/environs/imagemetadata"
+	"github.com/juju/juju/environs/simplestreams"
+	"github.com/juju/utils"
+)
+
+func init() {
+	simplestreams.RegisterStructTags(Metadata{})
+}
+
+const (
+	// DataType is the simplestreams datatype.
+	DataType = "image-downloads"
+)
+
+// DefaultSource creates a new signed simplestreams datasource for use with the
+// image-downloads datatype.
+func DefaultSource() simplestreams.DataSource {
+	ubuntuImagesURL := imagemetadata.UbuntuCloudImagesURL + "/" + imagemetadata.ReleasedImagesPath
+	return newDataSourceFunc(ubuntuImagesURL)()
+}
+
+// NewDataSource returns a new simplestreams.DataSource from the provided
+// baseURL. baseURL MUST include the image stream.
+func NewDataSource(baseURL string) simplestreams.DataSource {
+	return newDataSourceFunc(baseURL)()
+}
+
+// NewDataSource returns a datasourceFunc from the baseURL provided.
+func newDataSourceFunc(baseURL string) func() simplestreams.DataSource {
+	return func() simplestreams.DataSource {
+		return simplestreams.NewURLSignedDataSource(
+			"ubuntu cloud images",
+			baseURL,
+			imagemetadata.SimplestreamsImagesPublicKey,
+			utils.VerifySSLHostnames,
+			simplestreams.DEFAULT_CLOUD_DATA,
+			true)
+	}
+}
+
+// Metadata models the inforamtion about a particular cloud image download
+// product.
+type Metadata struct {
+	Arch string `json:"arch,omitempty"`
+	// For testing.
```

Comment:
Please clarify why we are having this in production code. What testing and how will it be used...